cas-server-core-util/src/main/java/org/jasig/cas/util/http/SimpleHttpClientFactoryBean.java

```diff
@@ -428,6 +429,14 @@ public void setRedirectsEnabled(final boolean redirectsEnabled) {
     }
 
     /**
+     * Destroy.
+     */
+    @PreDestroy
+    public void destroy() {
+        this.executorService.shutdownNow();
```

Comment:
You're perfectly right. I read the line backwards. I'll see if I can pull it out and shut it down explicitly.